crates/core/src/sql/execute.rs

```diff
@@ -79,26 +75,38 @@ pub fn execute_single_sql(
 #[tracing::instrument(skip_all)]
 pub fn execute_sql(
     db: &RelationalDB,
-    tx: &mut MutTx,
     ast: Vec<CrudExpr>,
     query_debug_info: Option<&QueryDebugInfo>,
     auth: AuthCtx,
 ) -> Result<Vec<MemTable>, DBError> {
     let total = ast.len();
     let ctx = ExecutionContext::sql(db.address(), query_debug_info);
-    let p = &mut DbProgram::new(&ctx, db, tx, auth);
-    let q = Expr::Block(ast.into_iter().map(|x| Expr::Crud(Box::new(x))).collect());
-
     let mut result = Vec::with_capacity(total);
-    collect_result(&mut result, run_ast(p, q).into())?;
+    match ast.iter().all(|expr| matches!(expr, CrudExpr::Query(_))) {
```

Comment:
Probably want to check with Mario or Joshua on this one if you haven't already, simply because I'm just not familiar.

Reply:
Let me add them as reviewer, specially on this snippet @joshua-spacetime , @mamcx

Reply:
Yes this looks correct. `CrudExpr::Query(_)` only takes `SELECT..` expressions without mutate effects.

I suggest to put this logic in `CrudExpres` (similar to `CrudExpr::is_reads()`) instead of here. The idea is that here is an upper layer and all decisions should be inside the `vm` or `query` layer so is more visible to the people that do optimizations.

Reply:
Agree, that makes sense